src/sdk/namespace_fs.js

```diff
@@ -402,7 +422,11 @@ class NamespaceFS {
                         return; // not added
                     }
 
-                    if (!delimiter && r.common_prefix) {
+                    if (ent.name === HIDDEN_VERSIONS_PATH && r.common_prefix) {
+                        if (list_versions) {
+                            await this._list_version_dir_objects(versions, ent.name, dir_key);
```

Comment:
why is this new function needed? can't we re-use the existing flow of processing a dir? 
and versions will be pushed inside the existing results array instead of the new versions array, the usage of the dir_cache also already sorts the entries for us

Reply:
Here in this case, `.versions` dir is parsed based on every `process_entry()`. So the `versions` list can be pushed on top of  the `results` stack based on the `create_time` and then `process_entry()` which is the latest object is pushed to the stack.